### PR TITLE
Initial train script

### PR DESCRIPTION
**Pull request by perrychu**
Training script runs, training loss decreases

Todo:
- Inference loop / Validation score
- Preprocess audio - training runs decently fast, but could avoid re-processing mel spectrograms by preprocessing. This would also let us filter out short samples to address the following:
- Training set audio has varying length. Currently handling by clipping a random 100-frame section from the mel spectrogram. Longer lengths run into some samples not being long enough.

### CODE REVIEW

**Comment by perrychu** [comment]
Sample model (256-dim speaker embedding, 5 epochs) - tacotron2/speaker/saved_model.pt

Load using "speaker.train.load_model(path)"
Or run "speaker.train.check_model(path)" to test on a single batch. I get loss around 10 +/- 3

**Comment by richieyoum** [comment]
I'm getting warnings when loading model about mel filterbank having all zero values, is this related to the mel_length [here](https://github.com/richieyoum/voice_cloning/blob/ea9e302870ba90429819ea06af99202bfaed9636/tacotron2/speaker/data.py#L56)?
Screenshot from 2022-04-09 21-11-55

**Comment by richieyoum** [comment]
FYI for others running this script you'd have to first set your pythonpath (ie - `export PYTHONPATH=./tacotron2`)

**Comment by perrychu** [comment]
> I'm getting warnings when loading model about mel filterbank having all zero values, is this related to the mel_length [here]

I was seeing those warnings too. My assumption is it was because sometimes the randomly selected segment might be silent, but need to look into it more.

If so, then one way to reduce it would be to increase the mel_length (which is the length of the selected segment). However we'll need to do add pre-processing to filter out training samples which are shorter than the selected length.

**Comment by perrychu** [comment]
After digging in, I think we can ignore the warning. It is generated in a utility function which creates a matrix which converts a frequency spectrogram to mel scale. 

The warning is generated from default settings in the MFCC transform we use. It is saying that there is a mel channel which won't pick up any signal. However in our implementation we only use 40 channels (they are collapsed from default 128 in a later step). 

Function which throws the warning here: https://pytorch.org/audio/stable/_modules/torchaudio/functional/functional.html#melscale_fbanks

On a side note - Our speaker encoder is currently using MFCC (Mel-frequency cepstral coefficients) which is a decorrelated version of the mel spectrogram. Based on a few quick experiments the MFCC representation converges a lot better for speaker verification. However in the actual voice generation I believe we'll have to use Mel Spectrogram since the output is waveform rather than an embedding representation.
